library/public/camera.h

```diff
@@ -0,0 +1,97 @@
+/**
+ * @class   camera
+ * @brief   Abstract class to control a camera in a camera
+ *
+ * A class to control a camera in a window, use only after the first render of the window.
+ */
+
+#ifndef f3d_camera_h
+#define f3d_camera_h
+
+#include "export.h"
+
+#include <array>
+#include <string>
+#include <stdexcept>
+
+namespace f3d
+{
+class F3D_EXPORT camera
+{
+public:
+  class exception : public std::runtime_error
+  {
+  public:
+    exception(const std::string& what = "")
+      : std::runtime_error(what)
+    {
+    }
+  };
+
+  typedef std::array<double, 3> vector_t;
+  typedef std::array<double, 16> matrix_t;
+
+  //@{
+  /**
+   * Set/Get the canera parameters.
+   * Call render on the window before calling any of these methods, or they will throw an exception.
+   */
+  virtual void setPosition(const vector_t& pos) = 0;
+  virtual vector_t getPosition() = 0;
+  virtual void setFocalPoint(const vector_t& foc) = 0;
+  virtual vector_t getFocalPoint() = 0;
+  virtual void setViewUp(const vector_t& up) = 0;
+  virtual vector_t getViewUp() = 0;
+  virtual void setViewAngle(const double& angle) = 0;
+  virtual double getViewAngle() = 0;
+  //@}
+ 
+ //@{
+  /**
+   * TODO
+   * Call render on the window before calling any of these methods, or they will throw an exception.
+   */
+  virtual void setViewMatrix(const matrix_t& matrix) = 0;
+  virtual matrix_t getViewMatrix() = 0;
```

Comment:
is this what you need @Meakk ?

Reply:
Yes, I need this one, but probably other functions too.

Reply:
Could you precise what you need so I can implement it ?

Reply:
Maybe not related to the camera, but I'll need way to get the world position from the position in screen space.  
But I'm not sure if I need it, I will investigate first.